lib/optimizely.rb

```diff
@@ -486,6 +485,12 @@ def get_feature_variable_for_type(feature_flag_key, variable_key, variable_type,
       # Returns nil if the feature flag or variable or user ID is empty
       #             in case of variable type mismatch
 
+      caller_method = caller_locations.first.label
```

Comment:
This seems a bit flaky. I'd rather keep the is_valid checks in the top level variable APIs for consistency since all the other public APIs have the same checks there.